python/packages/autogen-core/src/autogen_core/components/code_executor/_impl/utils.py

```diff
@@ -104,3 +106,46 @@ def infer_lang(code: str) -> str:
     except SyntaxError:
         # not a valid python code
         return "unknown"
+
+
```

Comment:
I have to say `create_virtual_environment` ended up being long, it's possible users would want to specify the parameters. However, if desired, we could modify this function to just take in `dir_path` and leave the rest of the values as defaults.

Reply:
I agree with this function being too long. Looking under the hood it is just a wrapper around `venv.EnvBuilder`, I suggest we remove this function and document the usage of `venv.EnvBuilder` directly

Reply:
Good idea!